python/cudf/cudf/core/udf/masked_typing.py

```diff
@@ -34,29 +41,60 @@
     types.NPTimedelta,
 )
 
+SUPPORTED_NUMPY_TYPES = (
+    NUMERIC_TYPES | DATETIME_TYPES | TIMEDELTA_TYPES | STRING_TYPES
+)
+supported_type_str = "\n".join(sorted(list(SUPPORTED_NUMPY_TYPES) + ["bool"]))
+MASKED_INIT_MAP: Dict[Any, Any] = {}
 
-class MaskedType(types.Type):
+
+def _format_error_string(err):
     """
-    A Numba type consisting of a value of some primitive type
-    and a validity boolean, over which we can define math ops
+    Wrap an error message in newlines and color it red.
     """
+    return "\033[91m" + "\n" + err + "\n" + "\033[0m"
 
-    def __init__(self, value):
-        # MaskedType in Numba shall be parameterized
-        # with a value type
-        if isinstance(value, SUPPORTED_NUMBA_TYPES):
-            self.value_type = value
+
+def _type_to_masked_type(t):
+    result = MASKED_INIT_MAP.get(t)
+    if result is None:
+        if isinstance(t, SUPPORTED_NUMBA_TYPES):
+            return t
         else:
             # Unsupported Dtype. Numba tends to print out the type info
             # for whatever operands and operation failed to type and then
             # output its own error message. Putting the message in the repr
             # then is one way of getting the true cause to the user
-            self.value_type = types.Poison(
-                "\n\n\n Unsupported MaskedType. This is usually caused by "
+            err = _format_error_string(
+                "Unsupported MaskedType. This is usually caused by "
                 "attempting to use a column of unsupported dtype in a UDF. "
-                f"Supported dtypes are {SUPPORTED_NUMBA_TYPES}"
+                f"Supported dtypes are:\n{supported_type_str}"
             )
-        super().__init__(name=f"Masked{self.value_type}")
+            return types.Poison(err)
+    else:
+        return result
+
+
+MASKED_INIT_MAP[types.pyobject] = types.Poison(
+    _format_error_string(
+        "strings_udf library required for usage of string dtypes "
+        "inside user defined functions. Try conda install strings_udf."
```

Comment:
I don't really like suggesting the `conda` command here. It would probably need a channel like `conda install -c rapidsai strings_udf` for most users (`rapidsai` is not a default channel), but that won't be accurate if the user is running a nightly build because they'd need `rapidsai-nightly`. It also wouldn't apply to users of other installation methods (e.g. from source). I would remove the `conda` command.
```suggestion
        "inside user defined functions."
```